Cabal/Distribution/Simple/ShowBuildInfo.hs

```diff
@@ -0,0 +1,153 @@
+-- |
+-- This module defines a simple JSON-based format for exporting basic
+-- information about a Cabal package and the compiler configuration Cabal
+-- would use to build it. This can be produced with the
+-- @cabal new-show-build-info@ command.
+--
+--
+-- This format is intended for consumption by external tooling and should
+-- therefore be rather stable. Moreover, this allows tooling users to avoid
+-- linking against Cabal. This is an important advantage as direct API usage
+-- tends to be rather fragile in the presence of user-initiated upgrades of
+-- Cabal.
+--
+-- Below is an example of the output this module produces,
+--
+-- @
+-- { "cabal-version": "1.23.0.0",
+--   "compiler": {
+--     "flavor": "GHC",
```

Comment:
What would tea-drinking-dragon say seeing this one!

Reply:
The `lib:Cabal` data type is called `CompilerFlavor`... in principle we can change the name for 3.0.